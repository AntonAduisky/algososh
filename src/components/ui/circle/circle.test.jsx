import renderer from 'react-test-renderer';

import { Circle } from './circle';


describe('Check props button', () => {
  it('Circle component are displayed correctly without props', () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle component letter-props are displayed correctly', () => {
    const tree = renderer.create(<Circle letter={'test text'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle component head-props are displayed correctly', () => {
    const tree = renderer.create(<Circle head={'24'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle component head-props are displayed correctly with React-element in props', () => {
    const tree = renderer.create(<Circle head={<Circle />} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle component tail-props are displayed correctly', () => {
    const tree = renderer.create(<Circle tail={'24'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle component tail-props are displayed correctly with React-element in props', () => {
    const tree = renderer.create(<Circle tail={<Circle />} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle component index-props are displayed correctly', () => {
    const tree = renderer.create(<Circle index={'24'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle component isSmall-props are displayed correctly', () => {
    const tree = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle component default-state-props are displayed correctly', () => {
    const tree = renderer.create(<Circle state={'default'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle component changing-state-props are displayed correctly', () => {
    const tree = renderer.create(<Circle state={'changing'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle component modified-state-props are displayed correctly', () => {
    const tree = renderer.create(<Circle state={'modified'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle component change-state-props are displayed correctly', () => {
    const tree = renderer.create(<Circle state={'change'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
