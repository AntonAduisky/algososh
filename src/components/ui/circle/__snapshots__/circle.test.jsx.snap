// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check props button Circle component are displayed correctly without props 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Check props button Circle component change-state-props are displayed correctly 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   change"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Check props button Circle component changing-state-props are displayed correctly 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Check props button Circle component default-state-props are displayed correctly 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Check props button Circle component head-props are displayed correctly 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    24
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Check props button Circle component head-props are displayed correctly with React-element in props 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
      data-cy="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-cy="circle-text"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Check props button Circle component index-props are displayed correctly 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    24
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Check props button Circle component isSmall-props are displayed correctly 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Check props button Circle component letter-props are displayed correctly 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    >
      test text
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Check props button Circle component modified-state-props are displayed correctly 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Check props button Circle component tail-props are displayed correctly 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    24
  </div>
</div>
`;

exports[`Check props button Circle component tail-props are displayed correctly with React-element in props 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
      data-cy="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-cy="circle-text"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;
